lib/koala/api/graph_batch_api.rb

```diff
@@ -138,6 +146,11 @@ def desired_component(component:, response:, headers:)
         # facebook returns the headers as an array of k/v pairs, but we want a regular hash
         when :headers then headers
         # (see note in regular api method about JSON parsing)
+        when :response
+          Koala::HTTPService::Response.new(
```

Comment:
let's return result which is defined above. And also you are mixing code and status in this initialization

Reply:
Good point @ylecuyer. We have use `result` defined above.